lib/server/services/ComponentService.js

```diff
@@ -745,23 +777,32 @@ module.exports = function(wilsonConfig, wilsonFrameworkConfig, logger, CacheServ
     componentJson.scripts = getComponentScripts(componentName, componentPath);
 
     // Add Service Dependency Scripts
-    var serviceScripts = [];
+    var serviceScripts  = [];
+
+    function getServiceDepPaths(service) {
+      if (buildTimeCache.serviceDeps[service]) { return buildTimeCache.serviceDeps[service]; }
+
+      // If we don't have this set of service paths cached find them
+      if (serviceLibrary[service]) {
+        // Get nested dependencies from the service library, and construct the service paths
+        var nestedDependencies = serviceLibrary[service].dependencies;
+        var servicePaths = _.pluck(_.pick(serviceLibrary, nestedDependencies), 'path');
+        servicePaths.push(serviceLibrary[service].path);
+        buildTimeCache.serviceDeps[service] = _.uniq(servicePaths);
+      }
+
+      return buildTimeCache.serviceDeps[service];
+    }
+
     _.each(componentJson.scripts, function (scriptPath) {      // This loop should be only over 1 js file
       // Get the service dependencies for the component
       serviceDependencies = getServiceDependencies(scriptPath, 'component');
 
-      // Add the scripts of each service dependency
       _.each(serviceDependencies, function (dependency) {
-        if (serviceLibrary[dependency]) {
-          // Get nested dependencies from the service library, and construct the service paths
-          var nestedDependencies = serviceLibrary[dependency].dependencies;
-          var servicePaths = _.pluck(_.pick(serviceLibrary, nestedDependencies), 'path');
-          serviceScripts = serviceScripts.concat(_.uniq(servicePaths));
-          serviceScripts.push(serviceLibrary[dependency].path);
-        }
+        serviceScripts = _.union(serviceScripts, getServiceDepPaths(dependency));
       });
     });
-    componentJson.scripts = _.uniq(serviceScripts.concat(componentJson.scripts));
+    componentJson.scripts = _.union(componentJson.scripts, serviceScripts);
```

Comment:
This for the specific case in the comment above:
<img width="587" alt="screen shot 2015-09-06 at 1 47 01 pm" src="https://cloud.githubusercontent.com/assets/1434287/9706582/d4a4eb9e-549d-11e5-8f51-ec9d82f21fe7.png">